src/target-i386/op_helper.c

```diff
@@ -4267,20 +4276,20 @@ void helper_fprem1(void) {
         else
             q = (signed long long int) dblq;
 
-        env->fpus &= (~0x4700);             /* (C3,C2,C1,C0) <-- 0000 */
+        FPUS_W(FPUS & (~0x4700));             /* (C3,C2,C1,C0) <-- 0000 */
                                             /* (C0,C3,C1) <-- (q2,q1,q0) */
-        env->fpus |= (q & 0x4) << (8 - 2);  /* (C0) <-- q2 */
-        env->fpus |= (q & 0x2) << (14 - 1); /* (C3) <-- q1 */
-        env->fpus |= (q & 0x1) << (9 - 0);  /* (C1) <-- q0 */
+        FPUS_W(FPUS | (q & 0x4) << (8 - 2));  /* (C0) <-- q2 */
```

Comment:
Make sure to put parenthesis around expressions, to make priority clear.

Reply:
Sure.